iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -528,6 +528,45 @@ private async Task OpenMethodsReceiverLinkAsync(AmqpIoTSession amqpIoTSession, s
             }
         }
 
+        public async Task DisableTwinPatchAsync(TimeSpan timeout)
+        {
+            if (Logging.IsEnabled)
+            {
+                Logging.Enter(this, timeout, $"{nameof(DisableTwinPatchAsync)}");
+            }
+
+            Debug.Assert(_twinSendingLink != null);
+            Debug.Assert(_twinReceivingLink != null);
+
+            try
+            {
+                ICollection<Task> tasks = new List<Task>();
+                if (_twinReceivingLink != null)
+                {
+                    tasks.Add(_twinReceivingLink.CloseAsync(timeout));
+                }
+
+                if (_twinSendingLink != null)
+                {
+                    tasks.Add(_twinSendingLink.CloseAsync(timeout));
```

Comment:
Is there a reason why we are closing the sending link as well? Disabling desired property updates should not have any impact on sending links.

Even for receiving links, I believe we are using the same receiving link to receive desired property updates and receive `GetTwinAsync()` on, right? Do we have proper locks to ensure that calling unsubscribe on twin updates and calling GetTwin() in parallel doesn't cause our internal state to get messed up?

Reply:
we open both sending and receiving link as part of enabling twin patch call so I suppose disabling should close both. 

Yes have a todo to make this thread safe.

Reply:
Twins would be a special case in that the same twin sending and receiving links are used for sending reported property updates, receiving desired property updated and retrieving the twin object.
If we close links on unsubscribe, we have to be very careful and put locks before we do that. That way it won't conflict with a separate twin operation attempting to open the links.
The other alternative is to simply stop processing unsolicited response.

What is the current behavior if we call GetTwinAsync() -> which will open both sending and receiving links, but don't subscribe to desired property updates. If I update a property from service, does the client receive it but then choses not to process it?

Reply:
Are you talking about Enabling Twin which in turn opens links? It already exists at both these places so should be the same flow.

Reply:
For unsubscribe - we will be closing sending and receiving links. Next call to use those links will open them.